Dodano funkcję makeInfinityAdder

JoAnna578 opened the pull request:
Zaimplementowano funkcję zwracającą adder, która sumuje liczby w łańcuchu wywołań i resetuje sumę przy wywołaniu bez argumentu.

JoAnna578 committed:
Dodano funkcję makeInfinityAdder

Zaimplementowano funkcję zwracającą adder, która sumuje liczby w łańcuchu wywołań i resetuje sumę przy wywołaniu bez argumentu.

mateacademy-ai-mentor commented on src/makeInfinityAdder.js:
```diff
@@ -4,7 +4,20 @@
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0; // zmienna w closure przechowująca sumę
```
The task requires the exported factory to be named `makeAdder`. The current implementation defines `makeInfinityAdder` instead; tests or consumers calling `makeAdder()` will not find the expected function. Rename the factory to `makeAdder` (or export it under the `makeAdder` name) so the module's API matches the specification in the task description .

mateacademy-ai-mentor commented on src/makeInfinityAdder.js:
```diff
@@ -4,7 +4,20 @@
 }
 
 module.exports = makeInfinityAdder;
+
```
The module export must expose the required function name. Currently the module is exporting `makeInfinityAdder` (see top-level declaration and export). Update the export so `module.exports` is the `makeAdder` function as required by the examples in the description. Prefer renaming the function to `makeAdder` to keep the implementation and API consistent.

mateacademy-ai-mentor commented on src/makeInfinityAdder.js:
```diff
@@ -4,7 +4,20 @@
-  // write code here
+  let sum = 0; // zmienna w closure przechowująca sumę
+
+  function adder(num) {
+    if (num === undefined) {   // wywołanie bez argumentu
+      const result = sum;       // zapisujemy wynik
+      sum = 0;                  // reset sumy
+      return result;            // zwracamy zgromadzoną sumę
+    }
+    sum += num;                 // dodajemy liczbę do sumy
+    return adder;               // zwracamy siebie do chainingu
+  }
+
+  return adder;
```
Implementation detail: distinguishing a no-argument call by checking `num === undefined` is acceptable and meets the requirement to detect the absence of arguments (the task forbids using default parameters for this purpose). The closure-based `sum` accumulator, returning `adder` for chaining, and resetting `sum` after a no-argument call are implemented correctly.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I am requesting changes because the exported API name does not match the requirement and will break consumers/tests expecting `makeAdder`. In `src/makeInfinityAdder.js`, the factory is defined and exported as `makeInfinityAdder` (top-level declaration and `module.exports`), but the task specifies the function must be named `makeAdder` and exported as such. Please rename the function to `makeAdder` and set `module.exports = makeAdder` (or export the existing implementation under the `makeAdder` name) so calls to `makeAdder()` work.

What you did well: the core logic is correct—closure-based accumulator, `num === undefined` to detect a no-arg call, returning the sum and resetting, and returning the adder for chaining. Minor improvement: consider aligning the filename to `makeAdder.js` for clarity. Once the export name is fixed, this should pass the tests. Keep up the good work on the implementation details!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JoAnna578 committed:
Zmiana nazwy funkcji na makeAdder

Zmieniono nazwę eksportowanej funkcji z makeInfinityAdder na makeAdder, aby spełniała wymagania specyfikacji i testów. Logika działania funkcji pozostaje bez zmian — closure, sumowanie liczb i reset przy wywołaniu bez argumentu.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

I am approving your solution. Your `makeAdder` factory returns an `adder` function that correctly accumulates values through chained single-argument calls and, on an empty call (detected via `num === undefined`), returns the accumulated sum and resets the internal total to 0 for reuse. This matches the required behavior (e.g., patterns like `adder(2)(3)...()`), and the export is correctly named `makeAdder`.

Minor suggestions (non-blocking):
- Consider renaming the file to `makeAdder.js` to align with the exported API and improve discoverability.
- As an optional style tweak, `arguments.length === 0` can be used for empty-call detection; your current `num === undefined` approach is also valid.

Nice work—clean, concise closure-based implementation with correct reset semantics.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
